core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -0,0 +1,357 @@
+   */
+  final def effectAsyncM[E, A](register: (IO[E, A] => Unit) => IO[Nothing, _]): IO[E, A] =
+    ZIO.effectAsyncM(register)
+
+  /** See [[scalaz.zio.ZIO.effectAsyncMaybe]]
+   */
+  final def effectAsyncMaybe[E, A](register: (IO[E, A] => Unit) => Option[IO[E, A]]): IO[E, A] =
+    ZIO.effectAsyncMaybe(register)
+
+  /** See [[scalaz.zio.ZIO.effectTotal]]
+   */
+  final def effectTotal[A](effect: => A): UIO[A] = ZIO.effectTotal(effect)
+
+  /** See [[scalaz.zio.ZIO.effectTotalWith]]
+   */
+  final def effectTotalWith[A](effect: Platform => A): UIO[A] = ZIO.effectTotalWith(effect)
+
+  /** See [[scalaz.zio.ZIO.fail]]
+   */
+  final def fail[E](error: E): IO[E, Nothing] = ZIO.fail(error)
+
+  /** See [[scalaz.zio.ZIO.flatten]]
+   */
+  final def flatten[E, A](zio: IO[E, IO[E, A]]): IO[E, A] =
+    ZIO.flatten(zio)
+
+  /** See [[scalaz.zio.ZIO.foldLeft]]
+   */
+  final def foldLeft[E, S, A](in: Iterable[A])(zero: S)(f: (S, A) => IO[E, S]): IO[E, S] =
+    ZIO.foldLeft(in)(zero)(f)
+
+  /** See [[scalaz.zio.ZIO.foreach]]
+   */
+  final def foreach[E, A, B](in: Iterable[A])(f: A => IO[E, B]): IO[E, List[B]] =
+    ZIO.foreach(in)(f)
+
+  /** See [[scalaz.zio.ZIO.foreachPar]]
+   */
+  final def foreachPar[E, A, B](as: Iterable[A])(fn: A => IO[E, B]): IO[E, List[B]] =
+    ZIO.foreachPar(as)(fn)
+
+  /** See [[scalaz.zio.ZIO.foreachParN]]
+   */
+  final def foreachParN[E, A, B](n: Long)(as: Iterable[A])(fn: A => IO[E, B]): IO[E, List[B]] =
+    ZIO.foreachParN(n)(as)(fn)
+
+  /** See [[scalaz.zio.ZIO.foreach_]]
+   */
+  final def foreach_[E, A](as: Iterable[A])(f: A => IO[E, _]): IO[E, Unit] =
+    ZIO.foreach_(as)(f)
+
+  /** See [[scalaz.zio.ZIO.foreachPar_]]
+   */
+  final def foreachPar_[E, A, B](as: Iterable[A])(f: A => IO[E, _]): IO[E, Unit] =
+    ZIO.foreachPar_(as)(f)
+
+  /** See [[scalaz.zio.ZIO.foreachParN_]]
+   */
+  final def foreachParN_[E, A, B](n: Long)(as: Iterable[A])(f: A => IO[E, _]): IO[E, Unit] =
+    ZIO.foreachParN_(n)(as)(f)
+
+  /** See [[scalaz.zio.ZIO.forkAll]]
+   */
+  final def forkAll[E, A](as: Iterable[IO[E, A]]): IO[Nothing, Fiber[E, List[A]]] =
+    ZIO.forkAll(as)
+
+  /** See [[scalaz.zio.ZIO.forkAll_]]
+   */
+  final def forkAll_[E, A](as: Iterable[IO[E, A]]): IO[Nothing, Unit] =
+    ZIO.forkAll_(as)
+
+  /** See [[scalaz.zio.ZIO.fromEither]]
+   */
+  final def fromEither[E, A](v: => Either[E, A]): IO[E, A] =
+    ZIO.fromEither(v)
+
+  /** See [[scalaz.zio.ZIO.fromFiber]]
+   */
+  final def fromFiber[E, A](fiber: => Fiber[E, A]): IO[E, A] =
+    ZIO.fromFiber(fiber)
+
+  /** See [[scalaz.zio.ZIO.fromFiberM]]
+   */
+  final def fromFiberM[E, A](fiber: IO[E, Fiber[E, A]]): IO[E, A] =
+    ZIO.fromFiberM(fiber)
+
+  /** See [[scalaz.zio.ZIO.fromFunction]]
+   */
+  final def fromFunction[A](f: Any => A): IO[Nothing, A] =
+    ZIO.fromFunction(f)
+
+  /** See [[scalaz.zio.ZIO.fromFunctionM]]
+   */
+  final def fromFunctionM[E, A](f: Any => IO[E, A]): IO[E, A] =
```

Comment:
This one only makes sense when `R` != `Any`, and it should be deleted from `IO`.